xrft/xrft.py

```diff
@@ -417,23 +449,29 @@ def dft(
 
     # verify even spacing of input coordinates
     delta_x = []
+    lag_x = []
     for d in dim:
         diff = _diff_coord(da[d])
         delta = np.abs(diff[0])
+        lag = _lag_coord(da[d])
         if not np.allclose(diff, diff[0], rtol=spacing_tol):
             raise ValueError(
                 "Can't take Fourier transform because "
                 "coodinate %s is not evenly spaced" % d
             )
         delta_x.append(delta)
+        lag_x.append(lag)
 
     if detrend:
         da = _apply_detrend(da, dim, axis_num, detrend)
 
     if window:
         da = _apply_window(da, dim)
 
-    f = fft_fn(da.data, axes=axis_num)
+    if true_phase:
+        f = fft_fn(np.fft.ifftshift(da.data, axes=axis_num), axes=axis_num)
```

Comment:
What happens here when `da.data` is a dask array? Is it computed eagerly?

Reply:
Fixing this is probably as simple as:
```suggestion
        f = fft_fn(fft.ifftshift(da.data, axes=axis_num), axes=axis_num)
```

Reply:
Thanks @rabernat !
I agree whith your suggestion (corrected in the new commit). I have done a small check on how it behaves with dask array and it seems to be ok (lazy computation). Is it everything good now ?

Reply:
@lanougue Could you also update `test_phase_preservation` to test for dask?

Reply:
@roxyboy : I don't understand what I should do on this test function. What I understand from @rabernat comment is that it should exist another test that checks if dft is actually computed lazily with dask arrays. Am I right ?
I do not beleive that it should be tested in test_true_phase_preservation().

Should we do another test that checks for lazy computations with dask arrays? If yes, I propose testing dtype of dft output when passing dask arrays as input.

Reply:
@lanougue Ok, considering that the test data in `test_phase_preservation` is one-dimensional, I guess it doesn't make sense to test `dask` there...

Reply:
I just addressed this in my latest push to @lanougue's branch.

We need to make sure that we explicitly check that computation remains lazy when calling `dft` with `check_phase=True`. It doesn't matter if Dask is realistically needed / reasonable in this particular test case. The laziness is the important property.